src/services/signedTransaction.ts

```diff
@@ -40,66 +40,44 @@ const submitToQueue = async (req: Request, res: Response) => {
 const submit = async (req: Request, res: Response) => {
   const buffer = Buffer.from(req.body.signedTx, "base64");
   const LOGGING_MSG_HOLDER: [null | string, null | string] = [null, null];
-  try {
-    const endpointResponse: any = await axios({
-      method: "post",
-      url: submissionEndpoint,
-      data: buffer,
-      headers: contentTypeHeaders,
-    });
+  let requestUrl = submissionEndpoint;
+  const headers: Record<string, string> = { ...contentTypeHeaders };
 
-    if (endpointResponse.status === 200) {
-      res.send([]);
-      return;
-    } else {
-      const { status, statusText, data } = endpointResponse || {};
-      throw Error(
-        `I did not understand the response from the submission endpoint: ${JSON.stringify(
-          {
-            status,
-            statusText,
-            data,
-            err: LOGGING_MSG_HOLDER[1],
-          }
-        )}`
-      );
-    }
-  } catch (error: any) {
-    const msg = `Error trying to send transaction: ${error} - ${JSON.stringify(
-      LOGGING_MSG_HOLDER
-    )}`;
-    throw Error(msg);
+  if (req.body.customNodeUrl) {
+    requestUrl = req.body.customNodeUrl;
+    delete headers.project_id;
   }
-};
-
-const submitToCustomURL = async (req: Request, res: Response) => {
-  const customNodeUrl = req.body.customNodeUrl;
-  // validate URL?
-  const buffer = Buffer.from(req.body.signedTx, "base64");
-  const LOGGING_MSG_HOLDER: [null | string, null | string] = [null, null];
 
   try {
     const endpointResponse: any = await axios({
       method: "post",
-      url: customNodeUrl,
+      url: requestUrl,
       data: buffer,
+      headers,
     });
 
     if (endpointResponse.status === 200) {
       res.send([]);
       return;
     } else {
+      const { status, statusText, data } = endpointResponse || {};
```

Comment:
Yeah I did use what was done before. But I made some changes, now in case of error from submission endpoint, I am returning the stringified response in error message.